### PR TITLE
Feature OSM->OSW

### PR DESCRIPTION
**Pull request by sujata-m**
Added OSM -> OSW conversion based on file type

1. If received `zip` file,  it will convert it to `xml` and uploaded to container
2. If received `pbf` file,  it will convert it to `zip` and uploaded to container 

Updated unit test cases

### CODE REVIEW

File: tests/integration_tests/test_osw_formatter_integration.py [hunk 1]
```diff
@@ -38,46 +39,75 @@ def setUpClass(cls):
...
+        message = QueueMessage.data_from({'message': '', 'data': self.test_data2})
+        formatter.publishing_topic.publish(data=message)
+        validation_topic = self.core.get_topic(topic_name=self.validation_topic_name)
+        async with validation_topic.subscribe(subscription='temp-validation-result', callback=subscribe_function):
+            await asyncio.sleep(0.5)  # Wait for callback
+            subscribe_function.assert_called_once()
+            formatter.start_listening.assert_called_once()
+
+    def test_zip_file_entity(self):
         test_file_url = 'https://tdeisamplestorage.blob.core.windows.net/osw/test_upload/osw.zip'
```
Comment (ayydevara) [inline tests/integration_tests/test_osw_formatter_integration.py:111]: are these files always available? If not, it is better to keep these files in this repo as test/resources

File: tests/integration_tests/test_osw_formatter_integration.py [hunk 2]
```diff
@@ -87,6 +117,16 @@ def test_file_entity(self):
         content = file.get_stream()
         self.assertTrue(content)
 
+    def test_pbf_file_entity(self):
+        test_file_url = 'https://tdeisamplestorage.blob.core.windows.net/osw/test_upload/wa.microsoft.osm.pbf'
```
Comment (ayydevara) [inline tests/integration_tests/test_osw_formatter_integration.py:121]: same question as above

**Review by ayydevara — APPROVED** [review]
While I comprehend the essence of these tests, it would significantly benefit code reviewers if there were a standardized pattern for tests, ensuring that the code's intent is readily apparent during the review.

The AAA pattern, denoting Arrange, Act, and Assert, serves as a widely adopted framework for composing unit tests. It furnishes a systematic approach to organizing your test code. Here's a breakdown of how you can format your Python tests using the AAA structure:

- **Arrange:** Establish the necessary preconditions and inputs for the test.
  
- **Act:** Execute the action or invoke the function intended for testing.
  
- **Assert:** Validate that the actual output or behavior aligns with the anticipated result.

Here is an example for this pattern can help review these tests easily

```py
import unittest

def add_numbers(a, b):
    return a + b

class TestAddNumbers(unittest.TestCase):

    def test_add_numbers_positive_case(self):
        # Arrange
        num1 = 5
        num2 = 10

        # Act
        result = add_numbers(num1, num2)

        # Assert
        self.assertEqual(result, 15, "Expected sum of 5 and 10 is 15")

    def test_add_numbers_negative_case(self):
        # Arrange
        num1 = -3
        num2 = 8

        # Act
        result = add_numbers(num1, num2)

        # Assert
        self.assertNotEqual(result, 10, "Expected sum of -3 and 8 is not 10")

if __name__ == '__main__':
    unittest.main()

```

**Review by sureshgaussian — APPROVED** [review]
I only ran unit tests, and made sure they pass. Didnt run the integration tests. Go ahead and merge.

File: tests/integration_tests/test_osw_formatter_integration.py (hunk 1 above)
Comment (sujata-m) [reply tests/integration_tests/test_osw_formatter_integration.py:111]: yes, its there

File: tests/integration_tests/test_osw_formatter_integration.py (hunk 2 above)
Comment (sujata-m) [reply tests/integration_tests/test_osw_formatter_integration.py:121]: yes, its there

**Pull request merged**